bookingBmodule.js

```diff
@@ -0,0 +1,117 @@
+    let dateToday = new Date().toJSON().slice(0, 10);
+
+    if (inputDate.value <= dateToday) {
+        window.alert("select a future date");
+    } else {
+        //to get available slots
+        bookingSlots(challengeDataTitle, challengeDataId, challengeDataMinParticipants, challengeDataMaxParticipants, inputDate);
+    }
+}
+
+//add secondModal to body
+function modalSection2(challengeDataTitle, challengeDataId, challengeDataMinParticipants, challengeDataMaxParticipants, inputDate, obj) {
+    const secondModal = createSecondModal(challengeDataTitle, challengeDataId, challengeDataMinParticipants, challengeDataMaxParticipants, inputDate, obj);
+    secondModal.style.display = "block";
+    const body = document.querySelector("body").appendChild(secondModal);
+}
+
+//create secondModal
+function createSecondModal(challengeDataTitle, challengeDataId, challengeDataMinParticipants, challengeDataMaxParticipants, inputDate, obj) {
+
+    const secondModal = document.createElement("section");
+    secondModal.setAttribute("class", "modal2");
+
+    const headline = document.createElement("h2");
+    headline.setAttribute("class", "modal2__headline");
+    headline.textContent = `Book room "${challengeDataTitle}" (step 2)`;
+
+    const inputLabel1 = document.createElement("label");
+    inputLabel1.setAttribute("class", "modal2__inputLabel1");
+    inputLabel1.setAttribute("for", "userName");
+    inputLabel1.textContent = "Name";
+
+    const inputName = document.createElement("input");
+    inputName.setAttribute("class", "modal2__inputName");
+    inputName.setAttribute("type", "text");
+    inputName.setAttribute("name", "userName");
+    inputName.setAttribute("required", "");
+
+    const inputLabel2 = document.createElement("label");
+    inputLabel2.setAttribute("class", "modal2__inputLabel2");
+    inputLabel2.setAttribute("for", "email");
+    inputLabel2.textContent = "E-mail";
+
+    const inputEmail = document.createElement("input");
+    inputEmail.setAttribute("class", "modal2__inputEmail");
+    inputEmail.setAttribute("type", "email");
+    inputEmail.setAttribute("name", "email");
+    inputEmail.setAttribute("required", "");
+
+    const slotLabel = document.createElement("label");
+    slotLabel.setAttribute("class", "modal2__slotLabel");
+    slotLabel.setAttribute("for", "slots");
+    slotLabel.textContent = "What time?";
+
+    const selectSlot = document.createElement("select");
+    selectSlot.setAttribute("class", "modal2__selectSlot");
+    selectSlot.setAttribute("type", "time");
+    selectSlot.setAttribute("name", "slots");
+    selectSlot.setAttribute("required", "");
+
+    for (let i = 0; i < obj.slots.length; i++) {
+        const slot = document.createElement("option")
+        slot.setAttribute("class", "modal2__selectSlot--slotTime");
+        slot.setAttribute("value", [i]);
```

Comment:
I think the expression you're using in the second parameter (`[i]`) maybe isn't what you want. Basically it creates an array, which will then be stringified somehow (I'm not sure how) in order for it to be able to fit in the value attribute of the `option` DOM element.

Instead, I would recommend just using `i`, or actually even better, use the slot itself since it's fairly machine-readable:

```js
slot.setAttribute('value', obj.slots[i]);
```